types/decimal_test.go

```diff
@@ -0,0 +1,69 @@
+package types_test
+
+import (
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestDecimal(t *testing.T) {
+	testCases := []struct {
+		name string
+		f    func(t *testing.T)
+	}{
+		{name: "TestDecimalToString", f: decimalStringTest},
+		{name: "TestDecimalToFloat64", f: decimalFloat64Test},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, tc.f)
+	}
+}
+
+func decimalStringTest(t *testing.T) {
+	testCases := []struct {
```

Comment:
I get `1.11111111E+2147483656` from:
```java
BigDecimal d = new BigDecimal(new BigInteger("111111111"), Integer.MIN_VALUE);
d.toString();
```
but `1.11111111E-2147483640` from:
```go
types.NewDecimal(big.NewInt(111_111_111), math.MinInt32).String()
```

Reply:
There were some errors in the implementation. I fixed them now, it should work correctly.  I 've ported `java.math.BigDecimal#layoutChars` implementation.